Infrastructure-Scripts/Set-AppRoleAssignments/Set-AppRoleAssignments.ps1

```diff
@@ -0,0 +1,142 @@
+<#
+    .SYNOPSIS
+    Grants app role assignments to the app registrations of the given Azure resources using managed identities
+
+    .DESCRIPTION
+    Grants app role assignments to the app registrations of the given Azure resources using managed identities
+    Configuration of app registrations and their app roles which have resources' managed identities granted access to is stored in the configuration file passed in
+
+    .PARAMETER AppRegistrationConfigurationFilePath
+    File path of the app registration configuration file
+
+    .PARAMETER AppServiceName
+    The name of the App Service to grant access to its relevant app registrations' roles.
+
+    .PARAMETER Tenant
+    Azure Tenant the app service lives in
+
+    .PARAMETER DryRun
+    Writes an output of the changes that would be made with no actual execution.
+
+    .EXAMPLE
+    .\Set-AppRoleAssignments.ps1 -AppRegistrationConfigurationFilePath "C:\config.json" -AppServiceName das-env-foobar-as -Tenant tenant.onmicrosoft.com -DryRun $true
+#>
+
+[CmdletBinding()]
+Param(
+    [Parameter(Mandatory = $true)]
+    [String]$AppRegistrationConfigurationFilePath,
+    [Parameter(Mandatory = $true)]
+    [String]$AppServiceName,
+    [Parameter(Mandatory = $true)]
+    [String]$Tenant,
+    [Parameter(Mandatory = $false)]
+    [bool]$DryRun = $true
+)
+
+$ErrorActionPreference = "Stop"
+Import-Module "$PSScriptRoot\tools\Helpers.psm1" -Force
+
+If ($DryRun) {
+    Write-Warning "Processing Dry Run"
+}
+
+try {
+    $AppRegistrationConfiguration = Get-Content -Path $AppRegistrationConfigurationFilePath -Raw | ConvertFrom-Json
+    $Environment = Get-Environment -AppServiceName $AppServiceName
+    $ResourceNamePrefix = "das-$Environment"
+    $AppServiceNameSuffix = $AppServiceName.Replace($ResourceNamePrefix, "")
+    $AppRegistrationsToProcess = $AppRegistrationConfiguration.configuration | Where-Object { $_.appRegistrationSuffix -match $AppServiceNameSuffix -or $_.appRoles.resourceNameSuffix -match $AppServiceNameSuffix }
+
+    if (!$AppRegistrationsToProcess) {
+        throw "No app registrations to process. Check app service name or update configuration."
```

Comment:
Worth including $AppServiceNameSuffix to help with debugging? For example it should be more obvious if bad values passed in and/or the transformation went wrong